### DISCUSSION TITLE
Safe Exam Browser not installed in Offline mode (Without Internet)

### DISCUSSION THREAD

**Discussion by Chris251290** [SafeExamBrowser/seb-win-refactoring#483] [Q&A] [answered]
Hi Team,

I have installed VC++ Redist x64 manully in offline mode and then I tried to install SEB 3.4.0. But, SEB not installed. It throws the error during installation.

Here the screenshots:
![SS1](https://user-images.githubusercontent.com/111885140/194496359-93fc620c-309c-43cd-857c-4815522905bb.png)
![SS2](https://user-images.githubusercontent.com/111885140/194496398-6c332175-3104-4a58-bcd7-4f02e25be3e4.png)

Here the log:
[Safe_Exam_Browser_20221007131938.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/9731774/Safe_Exam_Browser_20221007131938.log)

After connected the internet back and tried installing the SEB, Then only SEB installed successfully.

Here the screenshot:
![SS3](https://user-images.githubusercontent.com/111885140/194496582-eb644519-4c14-4608-9d60-0b2c5010f8cd.png)

 Note: (The requirement applications Dotnet framework latest version, VC++ redist x64 file and Microsoft edge web view2 also installed before started installing the SEB).

We need to know why SEB is not installed in offline ?

**Comment by dbuechel** [comment, marked as answer]
According to the installation log, the installation failed due to the WebView2 dependency:

```
[33A0:01B8][2022-10-07T13:19:58]i319: Applied execute package: WebView2_Bootstrapper, result: 0x80040801, restart: None
[33A0:01B8][2022-10-07T13:19:58]e000: Error 0x80040801: Failed to execute EXE package.
```

The setup bundle only contains a bootstrapper for the WebView2 dependency, i.e. you need a working internet connection as the bootstrapper will automatically search, download and install the latest available WebView2 build.

If you want to install everything offline, then you'd need to install all required dependencies as listed in the release notes or README here on GitHub first, and then install SEB using the setup (.msi).

**Reply by Chris251290** [reply]
Thank you so much, Damian. I have downloaded the SEB .msi x64 file from github. Now, I can install the SEB in Offline mode. After installed all the required files and SEB, I can run the exam.

Thanks for the timely help. Great.
